crates/consensus/beacon/src/engine/event.rs

```diff
@@ -12,6 +12,20 @@ pub enum BeaconConsensusEngineEvent {
     CanonicalBlockAdded(Arc<SealedBlock>, Duration),
     /// A canonical chain was committed, and the elapsed time committing the data
     CanonicalChainCommitted(Box<SealedHeader>, Duration),
+    /// The consensus engine is involved in live sync, and has specific progress
+    LiveSyncProgress(ConsensusEngineLiveSyncProgress),
     /// A block was added to the fork chain.
     ForkBlockAdded(Arc<SealedBlock>),
 }
+
+/// Progress of the consensus engine during live sync.
+#[derive(Clone, Debug)]
+pub enum ConsensusEngineLiveSyncProgress {
+    /// The consensus engine is downloading blocks on the network.
```

Comment:
```suggestion
    /// The consensus engine is downloading blocks from the network.
```

or is "on" correct here?